firmware/thruna/BMS/Src/App/App_Soc.c

```diff
@@ -1,5 +1,50 @@
 #include <math.h>
 #include "App_Soc.h"
+#include <stdint.h>
+#include <float.h>
+#include "App_SharedConstants.h"
+#include "App_SharedProcessing.h"
+#include "lut/App_Lut_CellVoltageToSoc.h"
+
+#define STATE_OF_HEALTH (1.0f)
+#define SERIES_ELEMENT_FULL_CHARGE_C (5.9f * 3600.0f * 3.0f * STATE_OF_HEALTH)
+#define MS_TO_S (0.001f)
+#define SOC_TIMER_DURATION (110U)
+
+float App_Soc_GetSocFromOcv(float voltage)
+{
+    uint8_t lut_index = 0;
+
+    while ((voltage > ocv_soc_lut[lut_index]) && (lut_index < V_TO_SOC_LUT_SIZE))
+    {
+        lut_index++;
+    }
+
+    if (lut_index == V_TO_SOC_LUT_SIZE)
+    {
+        // Ensures that the index is in the LUT range
+        lut_index--;
+    }
+    return LUT_BASE_SOC + lut_index * 0.5f;
+}
+
+float App_Soc_GetOcvFromSoc(float soc_percent)
+{
+    uint8_t lut_index = 0;
+
+    while ((LUT_BASE_SOC + lut_index * 0.5f < soc_percent) && (lut_index < V_TO_SOC_LUT_SIZE))
+    {
+        lut_index++;
+    }
+
+    if (lut_index == V_TO_SOC_LUT_SIZE)
+    {
+        // Ensures that the index is in the LUT range
+        lut_index--;
+    }
+
+    return ocv_soc_lut[lut_index];
+}
 
 ExitCode App_Soc_Vote(float max_abs_difference, float soc_1, float soc_2, float soc_3, float *result)
```

Comment:
What does this function do?

Reply:
This isn't actually used and I should probably remove it, it's leftover from Prom's implementation. But what it's meant to do is to compare 3 soc readings from EEPROM and determine if any of them are corrupt. I do similar functionality in App_Eeprom.c

Reply:
yep please remove if not used